File: backend/app/schemas/bank_reconciliation.py
"""Bank Reconciliation — Pydantic v2 Schemas"""
from __future__ import annotations
from datetime import date, datetime
from typing import Any, Dict, List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.models.bank_reconciliation import (
    BRAccountType, BRImportSource, BRStatementStatus, BRLineMatchStatus,
    BRMatchMethod, BRERPTransactionType, BRAdjustmentType, BRRuleConditionType,
    BRAIAgentType, BRAIRecStatus,
)


# ─── Bank Account ─────────────────────────────────────────────────────────────

class BRBankAccountCreate(BaseModel):
    bank_name:                str
    branch_name:              Optional[str]  = None
    account_name:             str
    account_number_masked:    Optional[str]  = None
    currency:                 str            = "KES"
    country:                  str            = "KE"
    account_type:             BRAccountType  = BRAccountType.CURRENT
    reconciliation_enabled:   bool           = True
    mobile_money_flag:        bool           = False
    mpesa_till_or_paybill_no: Optional[str]  = None
    gl_account_id:            Optional[UUID] = None
    cash_account_id:          Optional[UUID] = None
    active:                   bool           = True
    notes:                    Optional[str]  = None

class BRBankAccountUpdate(BaseModel):
    bank_name:                Optional[str]          = None
    branch_name:              Optional[str]          = None
    account_name:             Optional[str]          = None
    account_number_masked:    Optional[str]          = None
    reconciliation_enabled:   Optional[bool]         = None
    mobile_money_flag:        Optional[bool]         = None
    mpesa_till_or_paybill_no: Optional[str]          = None
    gl_account_id:            Optional[UUID]         = None
    cash_account_id:          Optional[UUID]         = None
    active:                   Optional[bool]         = None
    notes:                    Optional[str]          = None

class BRBankAccountOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id:                       UUID
    bank_name:                str
    branch_name:              Optional[str]
    account_name:             str
    account_number_masked:    Optional[str]
    currency:                 str
    country:                  str
    account_type:             BRAccountType
    reconciliation_enabled:   bool
    mobile_money_flag:        bool
    mpesa_till_or_paybill_no: Optional[str]
    gl_account_id:            Optional[UUID]
    cash_account_id:          Optional[UUID]
    active:                   bool
    notes:                    Optional[str]
    created_at:               datetime


# ─── Statement ────────────────────────────────────────────────────────────────

class BRStatementCreate(BaseModel):
    bank_account_id:    UUID
    statement_date:     date
    period_start_date:  date
    period_end_date:    date
    opening_balance:    float = 0.0
    closing_balance:    float = 0.0
    statement_currency: str   = "KES"
    import_source:      BRImportSource = BRImportSource.MANUAL
    import_file_name:   Optional[str]  = None
    notes:              Optional[str]  = None
    imported_by:        Optional[str]  = None

class BRStatementOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id:                 UUID
    statement_no:       str
    bank_account_id:    UUID
    statement_date:     date
    period_start_date:  date
    period_end_date:    date
    opening_balance:    float
    closing_balance:    float
    statement_currency: str
    import_source:      BRImportSource
    import_file_name:   Optional[str]
    status:             BRStatementStatus
    imported_by:        Optional[str]
    imported_at:        Optional[datetime]
    closed_by:          Optional[str]
    closed_at:          Optional[datetime]
    notes:              Optional[str]
    total_credits:      float
    total_debits:       float
    total_lines:        int
    created_at:         datetime


# ─── Statement Line ───────────────────────────────────────────────────────────

class BRStatementLineCreate(BaseModel):
    transaction_date:      date
    value_date:            Optional[date]  = None
    external_reference:    Optional[str]   = None
    bank_reference:        Optional[str]   = None
    narration:             Optional[str]   = None
    counterparty_name:     Optional[str]   = None
    amount:                float
    debit_credit_indicator: str            = "C"
    running_balance:       Optional[float] = None

class BRStatementLineOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id:                      UUID
    statement_id:            UUID
    line_no:                 int
    transaction_date:        date
    value_date:              Optional[date]
    external_reference:      Optional[str]
    bank_reference:          Optional[str]
    narration:               Optional[str]
    counterparty_name:       Optional[str]
    amount:                  float
    debit_credit_indicator:  str
    running_balance:         Optional[float]
    normalized_reference:    Optional[str]
    normalized_narration:    Optional[str]
    match_status:            BRLineMatchStatus
    matched_amount:          float
    unmatched_amount:        float
    reconciliation_notes:    Optional[str]
    is_duplicate_suspected:  bool
    created_at:              datetime


# ─── Match ────────────────────────────────────────────────────────────────────

class BRMatchCreate(BaseModel):
    statement_line_id:    UUID
    erp_transaction_type: BRERPTransactionType
    erp_transaction_id:   Optional[UUID] = None
    erp_reference:        Optional[str]  = None
    erp_transaction_date: Optional[date] = None
    erp_amount:           Optional[float] = None
    erp_description:      Optional[str]  = None
    matched_amount:       float
    match_method:         BRMatchMethod  = BRMatchMethod.MANUAL
    confidence_score:     Optional[float] = None
    notes:                Optional[str]  = None
    created_by:           Optional[str]  = None

class BRMatchOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id:                    UUID
    statement_line_id:     UUID
    erp_transaction_type:  BRERPTransactionType
    erp_transaction_id:    Optional[UUID]
    erp_reference:         Optional[str]
    erp_transaction_date:  Optional[date]
    erp_amount:            Optional[float]
    erp_description:       Optional[str]
    matched_amount:        float
    match_method:          BRMatchMethod
    confidence_score:      Optional[float]
    is_active:             bool
    created_by:            Optional[str]
    approved_by:           Optional[str]
    approved_at:           Optional[datetime]
    notes:                 Optional[str]
    created_at:            datetime


# ─── Rule ─────────────────────────────────────────────────────────────────────

class BRRuleCreate(BaseModel):
    bank_account_id:         Optional[UUID] = None
    rule_name:               str
    condition_type:          BRRuleConditionType
    condition_value:         str
    target_transaction_type: BRERPTransactionType
    auto_apply:              bool  = True
    confidence_weight:       float = 0.8
    priority:                int   = 100
    active:                  bool  = True
    notes:                   Optional[str] = None

class BRRuleUpdate(BaseModel):
    rule_name:               Optional[str]                  = None
    condition_value:         Optional[str]                  = None
    target_transaction_type: Optional[BRERPTransactionType] = None
    auto_apply:              Optional[bool]                 = None
    confidence_weight:       Optional[float]                = None
    priority:                Optional[int]                  = None
    active:                  Optional[bool]                 = None
    notes:                   Optional[str]                  = None

class BRRuleOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id:                      UUID
    bank_account_id:         Optional[UUID]
    rule_name:               str
    condition_type:          BRRuleConditionType
    condition_value:         str
    target_transaction_type: BRERPTransactionType
    auto_apply:              bool
    confidence_weight:       float
    priority:                int
    active:                  bool
    notes:                   Optional[str]
    created_at:              datetime


# ─── Adjustment ───────────────────────────────────────────────────────────────

class BRAdjustmentCreate(BaseModel):
    statement_id:      UUID
    statement_line_id: Optional[UUID]       = None
    adjustment_type:   BRAdjustmentType
    amount:            float
    description:       str
    gl_account_id:     Optional[UUID]       = None
    created_by:        Optional[str]        = None

class BRAdjustmentOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id:                UUID
    statement_id:      UUID
    statement_line_id: Optional[UUID]
    adjustment_type:   BRAdjustmentType
    amount:            float
    description:       str
    gl_account_id:     Optional[UUID]
    auto_created:      bool
    posted:            bool
    posted_by:         Optional[str]
    posted_at:         Optional[datetime]
    created_by:        Optional[str]
    created_at:        datetime


# ─── AI Recommendation ────────────────────────────────────────────────────────

class BRAIRecOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id:               UUID
    statement_id:     Optional[UUID]
    statement_line_id: Optional[UUID]
    agent_type:       BRAIAgentType
    title:            str
    explanation:      Optional[str]
    impact_summary:   Optional[str]
    suggested_action: Optional[str]
    priority:         str
    confidence_score: Optional[float]
    status:           BRAIRecStatus
    actioned_by:      Optional[str]
    actioned_at:      Optional[datetime]
    created_at:       datetime


# ─── Import / CSV ─────────────────────────────────────────────────────────────

class CSVImportRequest(BaseModel):
    bank_account_id:    UUID
    statement_date:     date
    period_start_date:  date
    period_end_date:    date
    opening_balance:    float
    closing_balance:    float
    statement_currency: str  = "KES"
    import_source:      BRImportSource = BRImportSource.CSV
    imported_by:        Optional[str]  = None
    lines:              List[BRStatementLineCreate]


# ─── Match requests ───────────────────────────────────────────────────────────

class ManualMatchRequest(BaseModel):
    erp_transaction_type: BRERPTransactionType
    erp_transaction_id:   Optional[UUID] = None
    erp_reference:        Optional[str]  = None
    erp_transaction_date: Optional[date] = None
    erp_amount:           Optional[float] = None
    erp_description:      Optional[str]  = None
    matched_amount:       float
    notes:                Optional[str]  = None
    created_by:           Optional[str]  = None

class SplitMatchRequest(BaseModel):
    matches: List[ManualMatchRequest]

class AutoMatchResult(BaseModel):
    line_id:         UUID
    candidates:      List[Dict[str, Any]]
    best_confidence: float
    auto_applied:    bool


# ─── Dashboard / Reports ──────────────────────────────────────────────────────

class BRDashboard(BaseModel):
    total_accounts:          int
    active_statements:       int
    unmatched_items:         int
    partially_matched_items: int
    total_unmatched_amount:  float
    statements_in_recon:     int
    statements_locked:       int
    pending_ai_recs:         int
    recent_statements:       List[BRStatementOut] = Field(default_factory=list)

class BROpenItemRow(BaseModel):
    line_id:             UUID
    statement_no:        str
    bank_account_name:   str
    transaction_date:    date
    amount:              float
    debit_credit:        str
    unmatched_amount:    float
    match_status:        BRLineMatchStatus
    narration:           Optional[str]
    days_outstanding:    int

class BRBalanceSummaryRow(BaseModel):
    bank_account_id:   UUID
    bank_account_name: str
    currency:          str
    last_statement_no: Optional[str]
    statement_closing: float
    erp_book_balance:  float
    variance:          float
    status:            str

File: frontend/src/lib/bank_reconciliation.ts
import { apiClient } from "@/lib/api";

const BASE = "/api/v1/bank-reconciliation";

// ─── Enums ────────────────────────────────────────────────────────────────────

export type BRAccountType = "CURRENT" | "SAVINGS" | "MOBILE_MONEY" | "OVERDRAFT";
export type BRImportSource = "CSV" | "OFX" | "MANUAL" | "MPESA" | "API";
export type BRStatementStatus =
  | "DRAFT" | "IMPORTED" | "IN_RECONCILIATION" | "RECONCILED" | "CLOSED" | "LOCKED";
export type BRLineMatchStatus =
  | "UNMATCHED" | "SUGGESTED_MATCH" | "PARTIALLY_MATCHED" | "MATCHED"
  | "IGNORED" | "DUPLICATE_SUSPECTED" | "NEEDS_REVIEW";
export type BRMatchMethod = "AUTO" | "MANUAL" | "SPLIT" | "RULE_BASED";
export type BRERPTransactionType =
  | "AR_PAYMENT" | "AP_PAYMENT" | "PURCHASE_PAYMENT" | "EXPENSE_PAYMENT"
  | "PAYROLL" | "TRANSFER" | "MPESA_COLLECTION" | "MPESA_PAYOUT"
  | "BANK_CHARGE" | "INTEREST" | "REFUND" | "JOURNAL_ENTRY";
export type BRAdjustmentType =
  | "BANK_CHARGE" | "INTEREST" | "ROUNDING" | "FX_DIFFERENCE" | "WRITE_OFF" | "OTHER";
export type BRRuleConditionType =
  | "NARRATION_CONTAINS" | "REFERENCE_STARTS_WITH" | "REFERENCE_CONTAINS"
  | "AMOUNT_EQUALS" | "AMOUNT_RANGE" | "PAYEE_NAME_CONTAINS" | "ACCOUNT_SPECIFIC";
export type BRAIAgentType = "MATCH_CONFIDENCE" | "CASH_ANOMALY" | "RULE_OPTIMIZER";
export type BRAIRecStatus = "PENDING" | "ACCEPTED" | "REJECTED" | "DISMISSED";

// ─── Types ────────────────────────────────────────────────────────────────────

export interface BRBankAccount {
  id: string;
  bank_name: string;
  branch_name: string | null;
  account_name: string;
  account_number_masked: string | null;
  currency: string;
  country: string;
  account_type: BRAccountType;
  reconciliation_enabled: boolean;
  mobile_money_flag: boolean;
  mpesa_till_or_paybill_no: string | null;
  gl_account_id: string | null;
  cash_account_id: string | null;
  active: boolean;
  notes: string | null;
  created_at: string;
}

export interface BRStatement {
  id: string;
  statement_no: string;
  bank_account_id: string;
  statement_date: string;
  period_start_date: string;
  period_end_date: string;
  opening_balance: number;
  closing_balance: number;
  statement_currency: string;
  import_source: BRImportSource;
  import_file_name: string | null;
  status: BRStatementStatus;
  imported_by: string | null;
  imported_at: string | null;
  closed_by: string | null;
  closed_at: string | null;
  notes: string | null;
  total_credits: number;
  total_debits: number;
  total_lines: number;
  created_at: string;
}

export interface BRStatementLine {
  id: string;
  statement_id: string;
  line_no: number;
  transaction_date: string;
  value_date: string | null;
  external_reference: string | null;
  bank_reference: string | null;
  narration: string | null;
  counterparty_name: string | null;
  amount: number;
  debit_credit_indicator: string;
  running_balance: number | null;
  normalized_reference: string | null;
  normalized_narration: string | null;
  match_status: BRLineMatchStatus;
  matched_amount: number;
  unmatched_amount: number;
  reconciliation_notes: string | null;
  is_duplicate_suspected: boolean;
  created_at: string;
}

export interface BRMatch {
  id: string;
  statement_line_id: string;
  erp_transaction_type: BRERPTransactionType;
  erp_transaction_id: string | null;
  erp_reference: string | null;
  erp_transaction_date: string | null;
  erp_amount: number | null;
  erp_description: string | null;
  matched_amount: number;
  match_method: BRMatchMethod;
  confidence_score: number | null;
  is_active: boolean;
  created_by: string | null;
  approved_by: string | null;
  approved_at: string | null;
  notes: string | null;
  created_at: string;
}

export interface BRRule {
  id: string;
  bank_account_id: string | null;
  rule_name: string;
  condition_type: BRRuleConditionType;
  condition_value: string;
  target_transaction_type: BRERPTransactionType;
  auto_apply: boolean;
  confidence_weight: number;
  priority: number;
  active: boolean;
  notes: string | null;
  created_at: string;
}

export interface BRAdjustment {
  id: string;
  statement_id: string;
  statement_line_id: string | null;
  adjustment_type: BRAdjustmentType;
  amount: number;
  description: string;
  gl_account_id: string | null;
  auto_created: boolean;
  posted: boolean;
  posted_by: string | null;
  posted_at: string | null;
  created_by: string | null;
  created_at: string;
}

export interface BRAIRec {
  id: string;
  statement_id: string | null;
  statement_line_id: string | null;
  agent_type: BRAIAgentType;
  title: string;
  explanation: string | null;
  impact_summary: string | null;
  suggested_action: string | null;
  priority: string;
  confidence_score: number | null;
  status: BRAIRecStatus;
  actioned_by: string | null;
  actioned_at: string | null;
  created_at: string;
}

export interface BRDashboard {
  total_accounts: number;
  active_statements: number;
  unmatched_items: number;
  partially_matched_items: number;
  total_unmatched_amount: number;
  statements_in_recon: number;
  statements_locked: number;
  pending_ai_recs: number;
  recent_statements: BRStatement[];
}

export interface BROpenItem {
  line_id: string;
  statement_no: string;
  bank_account_name: string;
  transaction_date: string;
  amount: number;
  debit_credit: string;
  unmatched_amount: number;
  match_status: BRLineMatchStatus;
  narration: string | null;
  days_outstanding: number;
}

export interface BRBalanceSummary {
  bank_account_id: string;
  bank_account_name: string;
  currency: string;
  last_statement_no: string | null;
  statement_closing: number;
  erp_book_balance: number;
  variance: number;
  status: string;
}

// ─── Color Helpers ────────────────────────────────────────────────────────────

export const STATUS_COLOR: Record<BRStatementStatus, string> = {
  DRAFT:             "bg-gray-100 text-gray-600",
  IMPORTED:          "bg-blue-100 text-blue-700",
  IN_RECONCILIATION: "bg-yellow-100 text-yellow-700",
  RECONCILED:        "bg-green-100 text-green-700",
  CLOSED:            "bg-purple-100 text-purple-700",
  LOCKED:            "bg-gray-700 text-white",
};

export const LINE_STATUS_COLOR: Record<BRLineMatchStatus, string> = {
  UNMATCHED:           "bg-red-100 text-red-700",
  SUGGESTED_MATCH:     "bg-yellow-100 text-yellow-700",
  PARTIALLY_MATCHED:   "bg-orange-100 text-orange-700",
  MATCHED:             "bg-green-100 text-green-700",
  IGNORED:             "bg-gray-100 text-gray-500",
  DUPLICATE_SUSPECTED: "bg-red-200 text-red-800",
  NEEDS_REVIEW:        "bg-purple-100 text-purple-700",
};

export const AI_AGENT_LABEL: Record<BRAIAgentType, string> = {
  MATCH_CONFIDENCE: "Match Confidence",
  CASH_ANOMALY:     "Cash Anomaly",
  RULE_OPTIMIZER:   "Rule Optimizer",
};

export const AI_PRIORITY_COLOR: Record<string, string> = {
  HIGH:   "bg-red-100 text-red-700",
  MEDIUM: "bg-yellow-100 text-yellow-700",
  LOW:    "bg-green-100 text-green-700",
};

export const ERP_TX_LABEL: Record<BRERPTransactionType, string> = {
  AR_PAYMENT:       "AR Payment",
  AP_PAYMENT:       "AP Payment",
  PURCHASE_PAYMENT: "Purchase Payment",
  EXPENSE_PAYMENT:  "Expense",
  PAYROLL:          "Payroll",
  TRANSFER:         "Transfer",
  MPESA_COLLECTION: "M-Pesa Collection",
  MPESA_PAYOUT:     "M-Pesa Payout",
  BANK_CHARGE:      "Bank Charge",
  INTEREST:         "Interest",
  REFUND:           "Refund",
  JOURNAL_ENTRY:    "Journal Entry",
};

// ─── API Client ───────────────────────────────────────────────────────────────

export const brApi = {
  // Dashboard
  getDashboard: () =>
    apiClient.get<BRDashboard>(`${BASE}/dashboard`).then((r) => r.data),

  // Bank accounts
  listAccounts: (activeOnly = true) =>
    apiClient.get<BRBankAccount[]>(`${BASE}/accounts`, { params: { active_only: activeOnly } }).then((r) => r.data),
  createAccount: (data: Partial<BRBankAccount>) =>
    apiClient.post<BRBankAccount>(`${BASE}/accounts`, data).then((r) => r.data),
  getAccount: (id: string) =>
    apiClient.get<BRBankAccount>(`${BASE}/accounts/${id}`).then((r) => r.data),
  updateAccount: (id: string, data: Partial<BRBankAccount>) =>
    apiClient.patch<BRBankAccount>(`${BASE}/accounts/${id}`, data).then((r) => r.data),

  // Statements
  listStatements: (params?: { bank_account_id?: string; status?: BRStatementStatus; limit?: number }) =>
    apiClient.get<BRStatement[]>(`${BASE}/statements`, { params }).then((r) => r.data),
  createStatement: (data: object) =>
    apiClient.post<BRStatement>(`${BASE}/statements`, data).then((r) => r.data),
  importStatement: (data: object) =>
    apiClient.post<BRStatement>(`${BASE}/statements/import`, data).then((r) => r.data),
  getStatement: (id: string) =>
    apiClient.get<BRStatement>(`${BASE}/statements/${id}`).then((r) => r.data),
  getLines: (id: string) =>
    apiClient.get<BRStatementLine[]>(`${BASE}/statements/${id}/lines`).then((r) => r.data),
  addLine: (id: string, data: object) =>
    apiClient.post<BRStatementLine>(`${BASE}/statements/${id}/lines`, data).then((r) => r.data),
  autoMatch: (id: string, params?: { confidence_threshold?: number; auto_apply_threshold?: number }) =>
    apiClient.post<object>(`${BASE}/statements/${id}/auto-match`, null, { params }).then((r) => r.data),
  closeStatement: (id: string, closedBy?: string) =>
    apiClient.post<BRStatement>(`${BASE}/statements/${id}/close`, null, { params: { closed_by: closedBy } }).then((r) => r.data),
  lockStatement: (id: string, lockedBy?: string) =>
    apiClient.post<BRStatement>(`${BASE}/statements/${id}/lock`, null, { params: { locked_by: lockedBy } }).then((r) => r.data),
  reopenStatement: (id: string, reopenedBy?: string) =>
    apiClient.post<BRStatement>(`${BASE}/statements/${id}/reopen`, null, { params: { reopened_by: reopenedBy } }).then((r) => r.data),
  getStatementReport: (id: string) =>
    apiClient.get<object>(`${BASE}/statements/${id}/report`).then((r) => r.data),
  listAdjustments: (id: string) =>
    apiClient.get<BRAdjustment[]>(`${BASE}/statements/${id}/adjustments`).then((r) => r.data),
  listStmtAIRecs: (id: string) =>
    apiClient.get<BRAIRec[]>(`${BASE}/statements/${id}/ai-recs`).then((r) => r.data),

  // Line actions
  manualMatch: (lineId: string, data: object) =>
    apiClient.post<BRMatch>(`${BASE}/lines/${lineId}/manual-match`, data).then((r) => r.data),
  splitMatch: (lineId: string, data: object) =>
    apiClient.post<BRMatch[]>(`${BASE}/lines/${lineId}/split-match`, data).then((r) => r.data),
  unmatch: (lineId: string) =>
    apiClient.post<BRStatementLine>(`${BASE}/lines/${lineId}/unmatch`).then((r) => r.data),
  ignoreLine: (lineId: string, notes?: string) =>
    apiClient.post<BRStatementLine>(`${BASE}/lines/${lineId}/ignore`, null, { params: { notes } }).then((r) => r.data),
  confirmSuggestion: (matchId: string, approvedBy?: string) =>
    apiClient.post<BRMatch>(`${BASE}/matches/${matchId}/confirm`, null, { params: { approved_by: approvedBy } }).then((r) => r.data),

  // Adjustments
  createAdjustment: (data: object) =>
    apiClient.post<BRAdjustment>(`${BASE}/adjustments`, data).then((r) => r.data),
  postAdjustment: (adjId: string, postedBy?: string) =>
    apiClient.post<BRAdjustment>(`${BASE}/adjustments/${adjId}/post`, null, { params: { posted_by: postedBy } }).then((r) => r.data),

  // Rules
  listRules: (bankAccountId?: string) =>
    apiClient.get<BRRule[]>(`${BASE}/rules`, { params: { bank_account_id: bankAccountId } }).then((r) => r.data),
  createRule: (data: object) =>
    apiClient.post<BRRule>(`${BASE}/rules`, data).then((r) => r.data),
  updateRule: (id: string, data: object) =>
    apiClient.patch<BRRule>(`${BASE}/rules/${id}`, data).then((r) => r.data),
  deleteRule: (id: string) =>
    apiClient.delete(`${BASE}/rules/${id}`),

  // Open items
  getOpenItems: (params?: { bank_account_id?: string; min_days?: number; limit?: number }) =>
    apiClient.get<BROpenItem[]>(`${BASE}/open-items`, { params }).then((r) => r.data),

  // Balance summary
  getBalanceSummary: () =>
    apiClient.get<BRBalanceSummary[]>(`${BASE}/balance-summary`).then((r) => r.data),

  // Reports
  reportUnreconciled: () =>
    apiClient.get<Record<string, unknown>[]>(`${BASE}/reports/unreconciled`).then((r) => r.data),
  reportAgedOpenItems: () =>
    apiClient.get<Record<string, unknown>[]>(`${BASE}/reports/aged-open-items`).then((r) => r.data),

  // AI
  runAI: (stmtId?: string) =>
    apiClient.post<{ recommendations_created: number }>(`${BASE}/ai/run`, null, { params: stmtId ? { stmt_id: stmtId } : undefined }).then((r) => r.data),
  listAIRecs: (params?: { status?: BRAIRecStatus }) =>
    apiClient.get<BRAIRec[]>(`${BASE}/ai/recs`, { params }).then((r) => r.data),
  actionAIRec: (id: string, status: BRAIRecStatus, actionedBy?: string) =>
    apiClient.post<BRAIRec>(`${BASE}/ai/recs/${id}/action`, null, { params: { status, actioned_by: actionedBy } }).then((r) => r.data),
};
